src/wire/mld.rs

```diff
@@ -386,6 +440,21 @@ impl<'a> Repr<'a> {
                 packet.set_nr_mcast_addr_rcrds(*nr_mcast_addr_rcrds);
                 packet.payload_mut().copy_from_slice(&data[..]);
             }
+            Repr::ReportRecordReprs(records) => {
+                packet.set_msg_type(Message::MldReport);
+                packet.set_msg_code(0);
+                packet.clear_reserved();
+                packet.set_nr_mcast_addr_rcrds(records.len() as u16);
+                let mut payload = packet.payload_mut();
+                for record in *records {
+                    if record.payload.len() == payload.len() {
+                        // TODO: handle the case where the payload sizes are different
+                        payload.copy_from_slice(record.payload);
+                    }
+                    record.emit(&mut AddressRecord::new_unchecked(&mut *payload));
+                    payload = &mut payload[record.buffer_len()..];
+                }
```

Comment:
Is this correct? I am not sure what you are trying to achieve with the if block, because I guess that the emit function of the record is going to override some parts of the payload variable.

Reply:
This was actually part of @jgallagher's original PR and, indeed, it can go. The source payload is always empty in the current implementation and, even if it weren't, the copied data would be overwritten when emitting the address record. I have just fixed this. Thanks for the code review!